Added ability to install Nagios from source.

kenhitchcock opened the pull request:
### What does this PR do?
Brief explanation of the code or documentation change you've made.
Role now includes the following.
 - A proper README file with variables explained, along with how the playbook and inventory should be configured to complete a successful deployment.

### How should this be tested?
Automated tests are preferred, but not always doable - especially for infrastructure. Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples)

Build a new RHEL7 system and register it to CDN
Create playbook and inventory according to README examples.
Run playbook against new VM built.
Ensure hostname of VM is resolvable then open browser to URL and use credentials set or default values.

### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.
resolves #<number>

None open, just a role that needed some refinement and enhancement.

### Other Relevant info, PRs, etc.
Please provide link to other PRs that may be related (blocking, resolves, etc. etc.)

### People to notify
cc: @redhat-cop/infra-ansible
@redhat-cop/infra-ansible

ostefek99 commented on roles/config-nagios-server/defaults/main.yml:
```diff
@@ -1 +1,46 @@
+nagios_epel_repo_rpm: epel-release-latest-7.noarch.rpm
+
+nagios_plugins_url: "https://github.com/nagios-plugins/nagios-plugins/archive"
+nagios_plugins_archive: "release-2.2.1.tar.gz"
```
Should the archive name be hardcoded?

I would use a variable just like in the case of `nagios-{{nagios_version}}.tar.gz`.

ostefek99 commented:
I would avoid using default values for versions of packages. They will become obsolete at some point. User should (IMO) always specify version that they want to use.

kenhitchcock commented:
@ostefek99 , thanks for the advice. I have taken note and removed the default values for the version. I have also added that the install will not run if the version values are not set. Makes perfect sense what you were saying and i agree 100%.

oybed commented:
FYI - lint checks have been enabled on this repo. Please resolve conflicts and push the latest.

kenhitchcock commented:
@ostefek99 @oybed 
The linting tests in the Pull request are confusing me. I have pulled out the tasks in the main.yml file and moved them to a configuration.yml file but still, the linting test say there is a conflict. The lines of conflict in the mail.yml don't exist in my pull request. Im stuck as to how to proceed with this one.

oybed commented:
@kenhitchcock just do a rebase of your fork/branch and you should be in good shape. BTW: the linting tests isn't the cause of the conflict, but rather that the file was updated/fixed in the upstream master and hence your fork/branch need to be brought up to be current with the latest. 

The main file that's causing the conflict is `roles/config-nagios-server/tasks/main.yml`, so just pay close attention to that one.
